### Урок 3. Описательная статистика. Качественные и количественные характеристики популяции. Графическое представление данных

#### 1. Даны значения зарплат из выборки выпускников: 100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150. Посчитать (желательно без использования статистических методов наподобие std, var, mean) среднее арифметическое, среднее квадратичное отклонение, смещенную и несмещенную оценки дисперсий для данной выборки.

In [66]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt


In [67]:
data_set = [100, 80, 75, 77, 89, 33, 45, 25,65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150]
print(data_set)

[100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150]


In [68]:
data_set.sort()
print(data_set)

[17, 24, 25, 30, 33, 45, 55, 57, 65, 65, 70, 75, 75, 77, 80, 84, 89, 90, 100, 150]


среднее арифметическое

$$M(X) = \frac{1}{n} \sum\limits_{i=1}^{n} x_i$$

In [69]:
17+24+25+30+33+45+55+57+65+65+70+75+75+77+80+84+89+90+100+150

1306

In [70]:
1306/20

65.3

In [71]:
# среднее арифметическое
mean_data_set = sum(data_set) / len(data_set)
print(f'Среднее арифметическое = {mean_data_set}')
    

Среднее арифметическое = 65.3


In [72]:
# среднее квадратичное отклонение


$$\sigma = \sqrt{\frac{\sum\limits_{i=1}^{n} (x_i - \overline{x})^2}{n}}$$

In [73]:
(17-65.3)**2+(24-65.3)**2+(25-65.3)**2+(30-65.3)**2+(33-65.3)**2+(45-65.3)**2+(55-65.3)**2+(57-65.3)**2+(65-65.3)**2+(65-65.3)**2+(70-65.3)**2+(75-65.3)**2+(75-65.3)**2+(77-65.3)**2+(80-65.3)**2+(84-65.3)**2+(89-65.3)**2+(90-65.3)**2+(100-65.3)**2+(150-65.3)**2

19002.2

In [74]:
19002.2 / 20

950.11

In [75]:
(950.11)**0.5

30.823854398825596

In [76]:
n = len(data_set)
s = 0
for i in data_set:
    s += (i - mean_data_set)**2   
sigma = (s / n)**0.5
print(f'Среднее квадратичное отклонение = {sigma}')

Среднее квадратичное отклонение = 30.823854398825596


**Дисперсия** равна среднему квадратичному отклонению, возведенному в квадрат:

$$\sigma^2 = \frac{\sum\limits_{i=1}^{n} (x_i - \overline{x})^2}{n}$$

In [77]:
D = sigma**2
print(f'Смещённая оценка дисперсии = {D}')

Смещённая оценка дисперсии = 950.11


**несмещенная оценка дисперсии**:
    
$$\sigma^2_{несмещ.} = \frac{\sum\limits_{i=1}^{n} (x_i - \overline{x})^2}{n - 1}$$

In [78]:
n = len(data_set)
s = 0
for i in data_set:
    s += (i - mean_data_set)**2   
sigma_2 = (s / (n - 1)) # НЕ взял n - 1 в скобки, из-за этого неверно посчиталось
print(f'Несмещённая оценка дисперсии = {sigma_2}')

Несмещённая оценка дисперсии = 1000.1157894736842


**Ответ:** 

Среднее арифметическое = 65.3

Среднее квадратичное отклонение = 30.823854398825596

Смещённая оценка дисперсии = 950.11

Несмещённая оценка дисперсии = 1000.11

#### 2. В первом ящике находится 8 мячей, из которых 5 - белые. Во втором ящике - 12 мячей, из которых 5 белых. Из первого ящика вытаскивают случайным образом два мяча, из второго - 4. Какова вероятность того, что 3 мяча белые?

In [79]:
from math import factorial

In [80]:
def comb(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

In [81]:
# вариант 1: 2 б. из первого ящика + 1 белый из второго:
p1 = (5/8*4/7) * ((5/12*7/11*6/10*5/9) + (7/12*5/11*6/10*5/9) + (7/12*6/11*5/10*5/9) + (7/12*6/11*5/10*5/9))
p1

0.1262626262626263

In [82]:
# вариант 2: 1 белый из первого ящика + 2 белых из второго:
p2 = ((5/8 * 3/7) + (3/8 * 5/7)) * ((5/12 * 4/11 * 7/10 * 6/9) + (5/12 * 7/11 * 4/10 * 6/9 ) + (7/12 * 5/11 * 6/10 * 4/9) + (7/12 * 6/11 * 5/10 * 4/9) + (5/12 * 7/11 * 6/10 * 4/9) + (7/12 * 5/11 * 4/10 * 6/9))
p2

#НЕ указал 2 варианта

0.22727272727272727

In [83]:
p2_c = ((comb(5,1) * comb(3,1)) / comb (8,2)) * ((comb(5,2) * comb(7,2)) / comb (12,4))
p2_c

0.22727272727272727

In [84]:
#варинт 3: 0 белых изи первого + 3 белых второго:
p3 = (3/8 * 2/7) * ((5/12*4/11*3/10*7/9)+(5/12*4/11*7/10*3/9)+(5/12*7/11*4/10*3/9)+(7/12*5/11*4/10*3/9)) #ошибка была в неверном знаке
p3

0.01515151515151515

In [85]:
p_total = p1 + p2 + p3
print(f'Вероятность того, что 3 мяча белые = {p_total:.4f}')

Вероятность того, что 3 мяча белые = 0.3687


**Ответ:**

Вероятность того, что 3 мяча белые = 0.3687



#### 3. На соревновании по биатлону один из трех спортсменов стреляет и попадает в мишень. Вероятность попадания для первого спортсмена равна 0.9, для второго — 0.8, для третьего — 0.6. Найти вероятность того, что выстрел произведен: a). первым спортсменом б). вторым спортсменом в). третьим спортсменом.

Воспользуемся формулой Байеса:

$$P(B\:|\:A) = \frac{P(B) \cdot P(A\:|\:B)}{P(A)}$$

In [86]:
p1 = 0.9
p2 = 0.8
p3 = 0.6
p_b = 1/3 # априорная вероятность выстрела для стрелков равна для всех стрелков
p_a = p_b*(p1 + p2 + p3)

In [87]:
p1_a = (p_b * p1) / p_a
print(f'Вероятность того, что выстрел произведен первым спортсменом = {p1_a}')

Вероятность того, что выстрел произведен первым спортсменом = 0.3913043478260869


In [88]:
p2_a = (p_b * p2) / p_a
print(f'Вероятность того, что выстрел произведен вторым спортсменом = {p2_a}')

Вероятность того, что выстрел произведен вторым спортсменом = 0.34782608695652173


In [89]:
p3_a = (p_b * p3) / p_a
print(f'Вероятность того, что выстрел произведен третьем спортсменом = {p3_a}')

Вероятность того, что выстрел произведен третьем спортсменом = 0.26086956521739124


**Ответ:** 

Вероятность того, что выстрел произведен первым спортсменом = 0.3913043478260869

Вероятность того, что выстрел произведен вторым спортсменом = 0.34782608695652173

Вероятность того, что выстрел произведен третьем спортсменом = 0.26086956521739124

#### 4. В университет на факультеты A и B поступило равное количество студентов, а на факультет C студентов поступило столько же, сколько на A и B вместе. Вероятность того, что студент факультета A сдаст первую сессию, равна 0.8. Для студента факультета B эта вероятность равна 0.7, а для студента факультета C - 0.9. Студент сдал первую сессию. Какова вероятность, что он учится: a). на факультете A б). на факультете B в). на факультете C?

Доли распределения на факультеты: X + X + 2X = 1, отсюда X = 0.25

In [90]:
pA = 0.8
pB = 0.7
pC = 0.9

p_b1 = 0.25
p_b2 = 0.25
p_b3 = 0.5
p_a = pA*p_b1 + pB*p_b2 + pC*p_b3


In [91]:
p1_a = (p_b1 * pA) / p_a
print(f'Вероятность того, студент из факультета A = {p1_a:.2f}')

Вероятность того, студент из факультета A = 0.24


In [92]:
p2_a = (p_b2 * pB) / p_a
print(f'Вероятность того, студент из факультета B = {p2_a:.2f}')

Вероятность того, студент из факультета B = 0.21


In [93]:
p3_a = (p_b3 * pC) / p_a
print(f'Вероятность того, студент из факультета C = {p3_a:.2f}')

Вероятность того, студент из факультета C = 0.55


**Ответ:**

Вероятность того, студент из факультета A = 0.24

Вероятность того, студент из факультета B = 0.21

Вероятность того, студент из факультета C = 0.55

#### 5. Устройство состоит из трех деталей. Для первой детали вероятность выйти из строя в первый месяц равна 0.1, для второй - 0.2, для третьей - 0.25. Какова вероятность того, что в первый месяц выйдут из строя: а). все детали б). только две детали в). хотя бы одна деталь г). от одной до двух деталей?

In [94]:
p1 = 0.1
p2 = 0.2
p3 = 0.25

In [95]:
a = p1*p2*p3
print(f'Вероятность того, что в первый месяц выйдут из строя все детали = {a:.4f}')

Вероятность того, что в первый месяц выйдут из строя все детали = 0.0050


In [96]:
b = (p1*p2*(1-p3)) + (p2*p3*(1-p1)) + (p1*p3*(1-p2))
print(f'Вероятность того, что в первый месяц выйдут из строя только две детали = {b:.2f}')

Вероятность того, что в первый месяц выйдут из строя только две детали = 0.08


In [97]:
q1 = 1 - p1
q2 = 1 - p2
q3 = 1 - p3

In [98]:
c = 1 - (q1 * q2 * q3)
print(f'Вероятность того, что в первый месяц выйдут из строя хотя бы одна деталь = {c:.2f}')
# неверное составил формулу

Вероятность того, что в первый месяц выйдут из строя хотя бы одна деталь = 0.46


In [99]:
d = c - a
print(f'Вероятность того, что в первый месяц выйдут из строя от одной до двух деталей = \
{d:.4f}')
#неверно составил формулу

Вероятность того, что в первый месяц выйдут из строя от одной до двух деталей = 0.4550


**Ответ:** 

Вероятность того, что в первый месяц выйдут из строя все детали = 0.0050

Вероятность того, что в первый месяц выйдут из строя только две детали = 0.08

Вероятность того, что в первый месяц выйдут из строя хотя бы одна деталь = 0.46

Вероятность того, что в первый месяц выйдут из строя от одной до двух деталей = 0.4550